transcom/mymove
# Premove Survey model, handler and migrations added to Shipments

## Description

This PR adds the migration to add pre move survey fields to the shipment table, model, and public handlers. Instead of keeping the internal and public API in the same handler file, I split out the public handlers into "shipments_public.go". The intention is to increase clarity and make navigation/comprehension easier - I'm interested in feedback on this and open to suggestions.

NOTE: the office (i.e. internal handlers) remain the same, meaning that the office shipment handler will not return a shipment with these newly added fields.

## Reviewer Notes

What's the best way to split up public and private handlers for the most clarity?

## Setup
1. 
```
make db_dev_reset 
make db_dev_migrate
bin/generate-test-data -scenario=7
```
-> to add fields to shipment table and generate shipments with associated shipment offers with "awarded" statuses.

2. Go to "tsplocal:3000", verify shipments appear in the queue.
3. Go to "tsplocal:3000/api/v1/docs". Get a shipment by id (find shipment id from shipment in queue list in previous step). Verify 200 response (you won't see newly added fields returned unless you manually add values to existing shipments).

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159742483) for this change
